enhancements/machine-api/machine-deletion-hooks.md

```diff
@@ -0,0 +1,525 @@
+---
+title: machine-deletion-hooks
+authors:
+  - "@JoelSpeed"
+  - "@michaelgugino"
+reviewers:
+  - "@enxebre"
+  - "@elmiko"
+  - "@ademicev"
+  - "@hexfusion"
+approvers:
+  - "@elmiko"
+  - "@hexfusion"
+  - "@deads2k"
+creation-date: 2021-08-10
+last-updated: 2021-11-11
+status: implementable
+see-also:
+  - [Cluster API Upstream Equivalent](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20200602-machine-deletion-phase-hooks.md)
+---
+
+# Machine Deletion Hooks
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Glossary
+
+### Lifecycle Hook
+A specific point in a machine's reconciliation lifecycle where execution of
+normal machine-controller behaviour is paused or modified.
+
+### Deletion Phase
+Describes when a machine has been marked for deletion but is still present
+in the API.  Various actions happen during this phase, such as draining a node,
+deleting an instance from a cloud provider, and deleting the node object.
```

Comment:
minor nit for consistency (we seem to use "cloud provider" or "infrastructure provider" interchangeably)
```suggestion
deleting an instance from an infrastructure provider, and deleting the node object.
```